src/tactic/tauto.lean

```diff
@@ -38,6 +38,13 @@ do hs ← local_context,
          | _ := failed
          end
 
+/-
+  The following definitions maintain a path compression datastructure, i.e. a forest such that:
+    - every node is the type of a hypothesis
+    - there is a edge between two nodes if and only if they are provably equivalent
```

Comment:
This is not a `iff`, it's only if. The edges are added when normalizing propositions.